hazelcast/src/main/java/com/hazelcast/map/impl/query/QueryDispatcher.java

```diff
@@ -125,6 +137,16 @@ protected QueryDispatcher(MapServiceContext mapServiceContext) {
         }
     }
 
+    // for rolling-upgrade compatibility, can be deleted in 4.0
```

Comment:
seems not only for rolling upgrade compatibility but also about creating an enterprise counterpart